src/Microsoft.Health.Fhir.Core/Features/Search/SearchHelperUtilities.cs

```diff
@@ -8,30 +8,45 @@
-        /// <see cref="ResourceSearchParameterStatus.Uri"/> and <see cref="ResourceSearchParameterStatus.LastUpdated"/>
-        /// values of each component. The same collection of search parameter status (irrespective of their order in the input)
+        /// Given a list of <see cref="SearchParameterInfo"/> calculates a hash using the
+        /// <see cref="SearchParameterInfo.Url"/>, <see cref="SearchParameterInfo.Type"/>,
+        /// <see cref="SearchParameterInfo.Expression"/>, <see cref="SearchParameterInfo.TargetResourceTypes"/>, and
+        /// <see cref="SearchParameterInfo.BaseResourceTypes"/>,
+        /// values of each component. The same collection of search parameter infos (irrespective of their order in the input)
         /// will return the same hash.
         /// </summary>
-        /// <param name="resourceSearchParameterStatus">A list of <see cref="ResourceSearchParameterStatus" /></param>
+        /// <param name="searchParamaterInfos">A list of <see cref="SearchParameterInfo" /></param>
         /// <returns>A hash based on the search parameter uri and last updated value.</returns>
-        public static string CalculateSearchParameterHash(IEnumerable<ResourceSearchParameterStatus> resourceSearchParameterStatus)
+        public static string CalculateSearchParameterHash(IEnumerable<SearchParameterInfo> searchParamaterInfos)
         {
-            EnsureArg.IsNotNull(resourceSearchParameterStatus, nameof(resourceSearchParameterStatus));
-            EnsureArg.IsGt(resourceSearchParameterStatus.Count(), 0, nameof(resourceSearchParameterStatus));
+            EnsureArg.IsNotNull(searchParamaterInfos, nameof(searchParamaterInfos));
+            EnsureArg.IsGt(searchParamaterInfos.Count(), 0, nameof(searchParamaterInfos));
 
             StringBuilder sb = new StringBuilder();
-            foreach (ResourceSearchParameterStatus searchParameterStatus in resourceSearchParameterStatus.OrderBy(x => x.Uri.ToString()))
+            foreach (SearchParameterInfo searchParamInfo in searchParamaterInfos.OrderBy(x => x.Url.ToString()))
             {
-                sb.Append(searchParameterStatus.Uri.ToString());
-                sb.Append(searchParameterStatus.LastUpdated.ToString());
+                sb.Append(searchParamInfo.Url.ToString());
+                sb.Append(searchParamInfo.Type.ToString());
+                sb.Append(searchParamInfo.Expression);
+
+                if (searchParamInfo.TargetResourceTypes != null &&
+                    searchParamInfo.TargetResourceTypes.Any())
+                {
+                    sb.Append(string.Join(null, searchParamInfo.TargetResourceTypes.OrderBy(s => s)));
+                }
+
+                if (searchParamInfo.BaseResourceTypes != null &&
+                    searchParamInfo.BaseResourceTypes.Any())
+                {
+                    sb.Append(string.Join(null, searchParamInfo.BaseResourceTypes.OrderBy(s => s)));
```

Comment:
Did we consider not allocating for the hash code? Using [`HashCode`](https://docs.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=dotnet-plat-ext-5.0)'s methods perhaps?

Reply:
I had not previously considered it, but I after looking at the Combine method, I am not sure it is a good match, as we have a varying number of items to combine, and the length can be more than what the different signatures of Combine allow for.

Reply:
For those scenarios, you can call HashCode.Add() like [here](https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/ActivityLink.netcoreapp.cs,24). Another approach is to call Combine() successively, but Add() likely gives better results.

Reply:
This was not a good suggestion. I did not realize that this was meant to be a proper cryptographic hash and not a HashCode implementation. Apologies for the detour.